### PR TITLE
[ENH] Render code blocks with language "tsv" as tables

### LINKED ISSUE #644: Harmonize .tsv examples (and their rendering)
Inspired by discussion in https://github.com/bids-standard/bids-specification/pull/619 I came to realization that there is no consistency ATM: .tsv "Text" examples come both in

<details>
<summary>proper TSV form</summary> 

```shell
$> git grep -e '        ' | grep '\.md'
src/02-common-principles.md:onset	duration	response_time	correct	stop_trial	go_trial
src/02-common-principles.md:200	200	0	n/a	n/a	n/a
src/03-modality-agnostic-files.md:filename	acq_time
src/03-modality-agnostic-files.md:func/sub-control01_task-nback_bold.nii.gz	1877-06-15T13:45:30
src/03-modality-agnostic-files.md:func/sub-control01_task-motor_bold.nii.gz	1877-06-15T13:55:33
src/04-modality-specific-files/08-genetic-descriptor.md:participant_id	age	sex	group	genetic_id	idh_mutation
src/04-modality-specific-files/08-genetic-descriptor.md:sub-control01	34	M	control	124587	yes
src/04-modality-specific-files/08-genetic-descriptor.md:sub-control02	12	F	control	548936	yes
src/04-modality-specific-files/08-genetic-descriptor.md:sub-patient01	33	F	patient	489634	no
src/05-derivatives/03-imaging.md:	    anat/
src/05-derivatives/03-imaging.md:		    sub-001_space-orig_probseg.nii.gz
src/05-derivatives/03-imaging.md:		    sub-001_space-orig_probseg.json
```
</details>
and more often in some shape of "replace tab with some spaces" - some times with nice layout into a tabular form, some times not really (even canonical example had no tabs but varying number of spaces used for delimiting, #643 ).  IMHO ideally

- markdown sources should use TABs where TABs are intended to be used
- rendering into HTML or PDF should do its best to signal that they are tabs while providing visually coherent display of the table.

I.e. ideally it should be a job of remake (some plugin) to do use TABS (ATM they are replaced by spaces by remake I guess), and just use HTML/CSS to provide nice visual alignment into the table (should be possible) while having TAB as the actual character so content could be copy pasted. Even though it is unlikely the actual entries would be copy pasted, I would expect some readers copy-paste headers (which also should be tab-delimited).  Could even use (although not really needed and overkill) smth like https://www.datatables.net/, where copy pasting a row from that fancy table does produce tab delimited entries:
```
$> xsel -o | sed -e 's, ,·,g' -e 's,\t,⇰,g'

Name⇰Position⇰Office⇰Age⇰%      
```

### LINKED ISSUE #1756: Add MACRO for every TSV add HTML tooltip with a warning if spaces were used
### Your idea

Inspired by
- https://github.com/bids-standard/bids-specification/pull/1749/files#r1545494653

The idea is that the warning in common principles would not be seen by majority of people who just end up at examples with TSVs which use spaces to improve indentation... Could be some kind of an alert sign with some modal window showing warning if pressed.

### LINKED ISSUE #1757: MACRO to "Render" TSV examples as a TABLE but tab-separated if copy/pasted
### Your idea

Pretty much an alternative to
- https://github.com/bids-standard/bids-specification/issues/1756

Here is an example which we came up with chatgpt: http://www.oneukrainian.com/tmp/tabs-demo.html . Try to copy/paste ;)

CSS styling could be adjusted so it is all tight and no borders etc, so just looks like a table of text.  The point is to address the shortcoming that now we need to indent TSV examples with spaces for "nice looking" rendering.

Concern originated
- https://github.com/bids-standard/bids-specification/pull/1749/files#r1545494653

Edit 1:
here is a better styled example: http://www.oneukrainian.com/tmp/tabs-demo2.html  . The "tricky" behavior I see. Whenever I just select on my linux laptop, which automagically copies to clipboard, I have desired part as tsv. When I click "ctrl-c" as to copy into clipboard explicitly - I seem to get the full table.  Could be treated as a feature ;)

Not sure yet what should be done for PDF -- probably just a note added ;-)

### PR DESCRIPTION
**Pull request by effigies**
Rendered: https://bids-specification--1794.org.readthedocs.build/en/1794/

This is another go at https://github.com/bids-standard/bids-specification/issues/644 along the lines of https://github.com/bids-standard/bids-specification/pull/1783#issuecomment-2062890572.

~~It's not working 100% for me, but I've maxed out my time on this. Pushed to upstream so others can take it up or open PRs against it.~~

The code for this does not need to live in bidsschematools, but that's the quickest way to get something into the import namespace.

cc @Remi-Gau 

- Closes #644
- Closes #1756 
- Closes #1757
- Closes #1783

### CODE REVIEW

**Comment by effigies** [comment]
Does not render for:

image


Does render, but with inserted `None` for:

image

Similarly successful but with additional `None` for the other files. The first one might be an effect of using it twice in a file? The two bugs may be related. I think the problem is probably upstream.

**Comment by Remi-Gau** [comment]
that's way quicker than anything I would have done

I can try tweaking this to see if I can make it work

**Comment by effigies** [comment]
FWIW I think this is not an `exclude-from-changelog` contribution, as it affects how to interpret what's seen in the spec.

**Comment by Remi-Gau** [comment]
> FWIW I think this is not an `exclude-from-changelog` contribution, as it affects how to interpret what's seen in the spec.

that's fair

File: src/modality-agnostic-files.md [hunk 1]
```diff
@@ -246,11 +246,11 @@ available").
...
-participant_id age sex handedness group
-sub-01 34 M right read
-sub-02 12 F right write
-sub-03 33 F n/a read
+```tsv
+participant_id	age	sex	handedness	group
+sub-01	34	M	right	read
+sub-02	12	F	right	write
+sub-03	33	F	n/a	read
 ```
```
Comment (effigies) [inline src/modality-agnostic-files.md:254]: https://bids-specification--1794.org.readthedocs.build/en/1794/modality-agnostic-files.html#participants-file

image

File: src/modality-agnostic-files.md [hunk 2]
```diff
@@ -320,13 +320,13 @@ and a guide for using macros can be found at
...
-sample-04 sub-02 tissue n/a
-sample-05 sub-02 tissue n/a
+```tsv
+sample_id	participant_id	sample_type	derived_from
+sample-01	sub-01	tissue	n/a
+sample-02	sub-01	tissue	sample-01
+sample-03	sub-01	tissue	sample-01
+sample-04	sub-02	tissue	n/a
+sample-05	sub-02	tissue	n/a
 ```
```
Comment (effigies) [inline src/modality-agnostic-files.md:330]: https://bids-specification--1794.org.readthedocs.build/en/1794/modality-agnostic-files.html#samples-file

image

File: src/modality-specific-files/positron-emission-tomography.md [hunk 3]
```diff
@@ -444,17 +444,17 @@ as shown in the following example.
...
+292	48.96	37.42	0.3149	0.2746
+602	39.84	32.05	0.1469	0.3548
+1248	37.38	31.52	0.073	0.444
+1785	36.40	28.83	0.078	0.429
+2390	33.13	26.32	0.061	0.453
+3059	30.83	25.22	0.049	0.473
+4196	27.28	21.98	0.036	0.503
+5407	22.70	19.49	0.032	0.523
+7193	19.71	15.70	0.02	0.559
 ```
```
Comment (effigies) [inline src/modality-specific-files/positron-emission-tomography.md:460]: https://bids-specification--1794.org.readthedocs.build/en/1794/modality-specific-files/positron-emission-tomography.html

image

**Review by effigies — COMMENTED** [review]
Links and screenshots of all affected locations.

File: mkdocs.yml [hunk 4]
```diff
@@ -104,7 +104,12 @@ extra_javascript:
 markdown_extensions:
   - toc:
       anchorlink: true
-  - pymdownx.superfences
+  - pymdownx.superfences:
+      preserve_tabs: true
+      custom_fences:
+        - name: tsv
+          class: tsv
+          format: !!python/name:bidsschematools.render.tsv.fence
```
Comment (yarikoptic) [inline mkdocs.yml:112]: image

**Review by yarikoptic — APPROVED** [review]
I guess original description should get this sentence crossed-out/removed?

> It's not working 100% for me, but I've maxed out my time on this. Pushed to upstream so others can take it up or open PRs against it.

I have tested a cut on linux producing tab separated representation somehow, so as to me it is everything anyone could hope for.

Well done @effigies . Thanks for posting all previews!

**Review by Remi-Gau — APPROVED** [review]
Heck yes!!!
Let's go (just because of the sheer number of things this closes)

**Comment by Remi-Gau** [comment]
- Probably we can consider this oses #1756

**Pull request merged**